src/DssSpell.sol

```diff
@@ -39,13 +43,71 @@ contract DssSpellAction is DssAction {
     //
     // uint256 internal constant X_PCT_RATE      = ;
 
-    address immutable internal ESM            = DssExecLib.getChangelogAddress("MCD_ESM");
-    address immutable internal RWA_LIQ_ORACLE = DssExecLib.getChangelogAddress("MIP21_LIQUIDATION_ORACLE");
+    uint256 internal constant FOUR_NINE_PCT_RATE = 1000000001516911765932351183;
+
+    uint256 internal constant MILLION = 10**6;
+    uint256 internal constant RAD     = 10**45;
 
     function actions() public override {
 
-        // ----- Additional ESM authorization fix -----
-        DssExecLib.authorize(RWA_LIQ_ORACLE, ESM);
+        uint256 Art;
```

Comment:
Sorry, I was confused here for a second regarding this. So, for spells, we have a checklist item `Ensure every spell variable is declared as public/internal`.

So we actually *do* explicitly declare visibility for **all** variables, whether it does anything or not. Originally this was a convention due to 0.6.12 and `nonPayable` being the default visibility. From the docs: ["The state mutability nonpayable is reflected in Solidity by not specifying a state mutability modifier at all."](https://docs.soliditylang.org/en/v0.8.19/abi-spec.html).

However, we are on 0.8.X now - the reason this is still important as a convention IMO is largely due to us wanting to avoid having external or public *variables* (constants are OK) on the top level of the spell, which cause the pause to not cast the spell (after elected to hat etc.).

In this case, [the archive pattern](https://github.com/makerdao/spells-mainnet/blob/fd9b613c2109867cc73d1da986befd8c33887008/archive/2023-03-11-DssSpell/DssSpell.sol#L151) doesn't declare visibility here, so this issue can be resolved, but I wanted to provide some context here.

Reply:
I'm still confused because those are **memory** variables, which exist only in the scope of the `action` method. I don't think there were ever any visibility modifiers for memory variables in any of the prior versions of Solidity.

I do understand that spells cannot contain state (storage) variables because they are [executed through a `delegatecall`](https://github.com/makerdao/ds-pause/blob/5e798dd96bfaac978cd9fe3c0259b486e8afd213/src/pause.sol#L151) and therefore are using `DSPauseProxy` storage and not their own.